public/js/controllers/testDetails.js

```diff
@@ -1,4 +1,15 @@
-CDash.controller('TestDetailsController',
+CDash.filter('colorize', ['$sce', function($sce) {
+  return function(input) {
+    // Fix escape characters
+    var pattern = /\[NON-XML-CHAR-0x1B\]/g;
+    input = input.replace(pattern, "\x1B");
+
+    // Now do the colors
+    var ansiUp = new AnsiUp;
+    return $sce.trustAsHtml(ansiUp.ansi_to_html(input));
+  };
+}])
```

Comment:
Please move this filter to a separate file (under `js/filters`).  Then we can re-use this code to colorize compiler output in `viewBuildError.php`.